src/pluggy/hooks.py

```diff
@@ -360,3 +361,142 @@ def __init__(self, namespace, name, opts):
         self.opts = opts
         self.argnames = ["__multicall__"] + list(self.argnames)
         self.warn_on_impl = opts.get("warn_on_impl")
+
```

Comment:
Yeah so this is basically reverting work to break this stuff *out of* `hooks.py` originally (see the history). The idea here was that we want a separate package/module space so we can keep track of and add new caller implementations. 

Imho putting this back into hooks is wrong and doesn't work towards that goal and we'll want to revert this again eventually anyway.